components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.hpp

```diff
@@ -63,6 +64,12 @@ class Nudging : public AtmosphereProcess
   // The name of the subcomponent
   std::string name () const { return "Nudging"; }
 
+  // Find the full filename list from patterns
+  std::vector<std::string> filename_glob(const std::vector<std::string>& patterns);
```

Comment:
These two functions do not change the state of the class. In fact, they don't even use any internals of the class. You could consider moving them inside `share/util/scream_util.hpp`. You could also simply declare them as local functions inside the cpp file, rather than methods. They don't conceptually belong to the public interface of the class...

Reply:
Thanks again for teaching me the basics.. will try to move them to scream_util.hpp

Reply:
Tbc, what you have is fine, and it works. It's just a matter of concept/design. Keeping large code bases tidy can help maintenance (especially when the maintainer is not the person that wrote the code).

Reply:
i like your suggestions, in fact I thought it would be more appropriate to put it in a more generic interface (but didn't know know where to put it at the time), since the glob function is a very generic function.. please check if the changes make sense